### ISSUE TITLE
Imbalanced classes

### ISSUE THREAD

**Issue by ddofer** [dnouri/nolearn#113]
Are there any plans to implement something for handling cases of imbalanced classes? 
(It'd be possible to use something external, I know, but I'd prefer not to risk mucking up the data, or use with anything that does "dataset augmentation".
 Ideal would be "weighted" costs/samples a la sklearn, but I have no idea if this is even makes sense (or is supported)!

Thanks :)

**Comment by run2** [comment]
Daniel, would love to hear from you on this. I have a regular binary classification problem with (highly) imbalanced class. If I chose regression=False the validation accuracy is calculated based on mean error. That starts with > 99% and never improves. Can we add some kind of auc eval ( in _create_iter_funcs ) ?

Regards
Deb

**Comment by dnouri** [comment]
@run2: There's actually a `custom_score` parameter where you can pass a tuple `(name, func)` where func is called with `(y_true, y_predict)`.  This will be then be displayed in the table.

**Comment by dnouri** [comment]
@ddofer: The same request came up a few times:
- Issue: Weighted loss function #109
- Pull request: Add ability to weight training samples and predefine training/validation splits #123

Note that I'm in the process of adding another interface for `objective`: https://github.com/dnouri/nolearn/pull/124

Lasagne's `lasagne.objectives.aggregate` has a `weights` argument that we could use, and that #123 uses.  I haven't looked at the problem in detail yet, though I was hoping that there could be a way to support weights (maybe not exactly like sklearn) without so many changes in the code.

**Comment by dnouri** [comment]
#123 and #124 have been merged.  Let's move the discussion to #109.

**Comment by BrianMiner** [comment]
@dnouri, Is y_true, y_predict that is used with cust_score training or validation?

**Comment by dnouri** [comment]
@BrianMiner validation

**Comment by ArpitaSTugave** [comment]
Hey how do I find training accuracy?

**Comment by ArpitaSTugave** [comment]
@dnouri how do I find training accuracy using custom_score?
